### DISCUSSION TITLE
Why limit the number of columns ?

### DISCUSSION THREAD

**Discussion by Euphillya** [plan-player-analytics/Plan#3818] [Q&A] [answered]
Hello, I saw on the [wiki](https://github.com/plan-player-analytics/Plan/wiki/APIv5---DataExtension-API#tableprovider) that:
- A player can have 4 columns
- The server can have 5 columns

Is there a specific reason for this limitation?

**Comment by AuroraLS3** [comment]
The data is stored in a fixed size database table so column count can't be unique per table. The column counts were chosen with 1920px display width in mind, since 5 columns can fill full width screen and 4 columns a 2/3rd size widget.

These were decided before the UI got option to hide columns from tables.

**Comment by Euphillya** [comment]
So, giving the developer the possibility to put more, would that be possible if we change the size of the table?
https://github.com/plan-player-analytics/Plan/blob/acd83a41c349dc18c361d8d336284954d1760375/Plan/api/src/main/java/com/djrapitops/plan/extension/table/Table.java#L56

**Reply by AuroraLS3** [reply]
It would need increasing the amount of database table columns and modifying all related queries.

There are 4 columns in the database for one column in the table (icon, formatter, title, cell_value) so storing a table with 5 columns takes 20 columns in the database schema.

**Reply by AuroraLS3** [reply, marked as answer]
It would be possible to have all cell values on individual rows with column index (for infinite columns), but it is more efficient for table data updates and queries to be able to change whole rows at a time rather than updating one cell at a time
